File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest=1
        n=len(nums)
        if n==0:
            return 0
        st=set()
        for i in range(n):
            st.add(nums[i])
        for it in st:
            if it-1 not in st:
                cnt=1
                x=it
                while x+1 in st:
                    x=x+1
                    cnt=cnt + 1 
                longest=max(longest,cnt)
        return longest
        


        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer>set=new HashSet<>();
        int max=0;
        for(int num:nums){
            set.add(num);
        }
        for(int num:set){
            if(set.contains(num-1)) continue;
            int currentLength=0;
            while(set.contains(num)){
                currentLength++;
                num+=1;
            }
            max=Math.max(max,currentLength);
        }
        return max;
    }
}